packages/solid-query/src/utils.ts

```diff
@@ -1,61 +1,33 @@
-import type {
-  SolidQueryKey,
-  SolidQueryFilters,
-  ParseFilterArgs,
-  ParseQueryArgs,
-} from './types'
-import type { QueryFunction, QueryOptions } from '@tanstack/query-core'
+import type { SolidQueryKey, SolidQueryFilters, ParseFilterArgs } from './types'
 
 export function isQueryKey(value: unknown): value is SolidQueryKey {
   return typeof value === 'function'
 }
 
 // The parseQuery Args functions helps normalize the arguments into the correct form.
 // Whatever the parameters are, they are normalized into the correct form.
-export function parseQueryArgs<
-  TOptions extends Omit<
-    QueryOptions<any, any, any, ReturnType<TQueryKey>>,
-    'queryKey'
-  > & {
-    queryKey?: TQueryKey
-  },
-  TQueryKey extends () => readonly unknown[] = SolidQueryKey,
->(
-  arg1: TQueryKey | TOptions,
-  arg2?: QueryFunction<any, ReturnType<TQueryKey>> | TOptions,
-  arg3?: TOptions,
-): ParseQueryArgs<TOptions, TQueryKey> {
-  if (!isQueryKey(arg1)) {
-    const { queryKey: solidKey, ...opts } = arg1 as any
-    if (solidKey) {
-      return {
-        ...opts,
-        queryKey: solidKey(),
-      }
+export function normalizeQueryOptions<T>(arg: T): T {
+  const { queryKey: solidKey, ...opts } = arg as any
+  if (solidKey) {
+    return {
+      ...opts,
+      queryKey: solidKey(),
     }
-    return arg1 as any
-  }
-
-  if (typeof arg2 === 'function') {
-    return { ...arg3, queryKey: arg1(), queryFn: arg2 } as any
   }
-
-  return { ...arg2, queryKey: arg1() } as any
+  return arg as any
 }
```

Comment:
not sure if this is still necessary. from what I can see, it tries to make the queryKey reactive ? I guess we'd now want that for the whole options object? Another question for @ardeora :)